source/adapters/cuda/image.cpp

```diff
@@ -814,11 +821,73 @@ UR_APIEXPORT ur_result_t UR_APICALL urBindlessImagesImageCopyExp(
         cpy_desc.Height = std::max(uint64_t{1}, copyExtent.height);
         cpy_desc.Depth = pImageDesc->arraySize;
         UR_CHECK_ERROR(cuMemcpy3DAsync(&cpy_desc, Stream));
+        cuStreamSynchronize(Stream);
       }
     } else {
-      /// imageCopyFlags == UR_EXP_IMAGE_COPY_FLAG_DEVICE_TO_DEVICE
```

Comment:
Could we keep this comment? Always helpful on an `else` like this.

Reply:
Not sure where the weird formatting came on the comment, but while we're here can you fix it to start with only two "//"?

Reply:
fixed.